src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java

```diff
@@ -86,16 +86,49 @@
  *    &lt;property name="format" value="^[a-z](_?[a-zA-Z0-9]+)*$"/&gt;
  * &lt;/module&gt;
  * </pre>
- *
+ * <p>Code Example:</p>
+ * <pre>
+ * class MyClass {
+ *   public void myMethod() {} // OK
+ *   public void MyMethod() {} // violation, name "MyMethod"
+ *                             // should match the pattern "^[a-z](_?[a-zA-Z0-9]+)*$"
+ * }
+ * </pre>
  * <p>
- * An example of how to configure the check to allow method names
- * to be equal to the residing class name is:
+ * An example of how to configure the check to allow method names to be equal to the
+ * residing class name is:
  * </p>
  * <pre>
  * &lt;module name="MethodName"&gt;
+ *    &lt;property name="format" value="^[a-zA-Z](_?[a-zA-Z0-9]+)*$"/&gt;
```

Comment:
I think I am missing your point. The property `allowClassName` allows the method name to be the same as the class. `format` is completely ignored in this situation.
Wouldn't a `format` property that doesn't match the class naming convention show more that `allowClassName` is overriding the format?

Reply:
Well, in a nutshell, it is because I think the class name's first letter should be capitalized. So I changed the format. If you think it doesn't matter, then I just start with the lower case in the class name. '''format''' property could be ignored in this situation.